.github/workflows/compile.yml

```diff
@@ -0,0 +1,13 @@
+on:
+  - push
+  - pull_request
+jobs:
+  compile-sketch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: arduino/compile-sketches@v1
+        with:
+          fqbn: "arduino:avr:uno"
+          sketch-paths: |
+            - src
```

Comment:
Thought: Is it worth adding the built firmware as an artifact?

Reply:
Not really the IDE will build it before flashing, this just proves it will build.